hollow/src/main/java/com/netflix/hollow/api/producer/validation/ObjectModificationValidator.java

```diff
@@ -0,0 +1,122 @@
+import com.netflix.hollow.api.producer.HollowProducer.Validator;
+import com.netflix.hollow.api.producer.HollowProducerListener.Status;
+import com.netflix.hollow.core.HollowConstants;
+import com.netflix.hollow.core.index.HollowPrimaryKeyIndex;
+import com.netflix.hollow.core.index.key.PrimaryKey;
+import com.netflix.hollow.core.read.engine.HollowTypeReadState;
+import com.netflix.hollow.core.read.engine.object.HollowObjectTypeReadState;
+import com.netflix.hollow.core.schema.HollowSchema.SchemaType;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.function.BiFunction;
+
+/**
+ * This validator runs a provided {@link BiFunction} on every instance of an object that was
+ * modified in a cycle. This "filter" function is called with the old GenericHollowObject as the
+ * first argument and the new GenericHollowObject as the second argument. If the function returns
+ * false, this validator is marked as failed and the producer cycle fails.
+ * Note that the filter function is not called with any objects that were added or removed.
+ * Also note that this validator can only be used on types that have a primary key.
+ */
+public class ObjectModificationValidator<T extends HollowObject> implements Validator {
+    private final String typeName;
+    private final BiFunction<T, T, Boolean> filter;
+
+    private SingleValidationStatus.SingleValidationStatusBuilder statusBuilder;
+
+    @SuppressWarnings("WeakerAccess")
+    public ObjectModificationValidator(String typeName, BiFunction<T, T, Boolean> filter) {
+		this.typeName = typeName;
+		this.filter = filter;
+        this.statusBuilder = SingleValidationStatus.builder(ObjectModificationValidator.class.getSimpleName())
+                .addAdditionalInfo("typeName", typeName);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+	public void validate(ReadState readState) throws ValidationException {
+        try {
+            HollowTypeReadState typeState = readState.getStateEngine().getTypeState(typeName);
+            if (typeState == null) {
+                throw new ValidationException("Cannot execute ObjectModificationValidator on missing type "
+                        + typeName);
+            } else if (typeState.getSchema().getSchemaType() != SchemaType.OBJECT) {
+                throw new ValidationException("Cannot execute ObjectModificationValidator on type " + typeName
+                        + " because it is not a HollowObjectSchema - it is a "
+                        + typeState.getSchema().getSchemaType());
+            }
+            BitSet latestOrdinals = typeState.getPopulatedOrdinals();
+            BitSet previousOrdinals = typeState.getPreviousOrdinals();
+            if (previousOrdinals.isEmpty()) {
+                statusBuilder.withMessage("Skipping " + ObjectModificationValidator.class.getSimpleName()
+                        + " because previous cycle has no " + typeName).withStatus(Status.SKIP);
```

Comment:
Arguably the validator is not skipping here and below when the ordinals are empty. There is nothing to validate so no work to do, where as skipping sort of implies "i could do work but i am choosing not to". IMHO the notion of skipping should be reconsidered.

Reply:
Paul's interpretation aligns with how I've been thinking of skipping. @lkancode is the original advocate on SKIP as a status, though, so should consult with her.

Reply:
Question for you guys - when _would_ you think a validator has been "skipped"?